training/src/main/java/wolox/training/services/OpenLibraryService.java

```diff
@@ -0,0 +1,93 @@
+package wolox.training.services;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import wolox.training.models.Book;
+
+@Service
+public class OpenLibraryService {
+
+    private final RestTemplate restTemplate;
+
+    public OpenLibraryService(RestTemplateBuilder restTemplateBuilder) {
+        this.restTemplate = restTemplateBuilder.build();
+    }
+
+    private String getNumberOfAlphanumericString(String alphanumericString) {
+        Pattern regex = Pattern.compile("\\d+");
+        Matcher numberFound = regex.matcher(alphanumericString);
+        while (numberFound.find()) {
+            return numberFound.group(0);
+        }
+        return "0";
+    }
+
+
+    public Book bookInfo(String isbn) throws ParseException, JsonProcessingException {
+        Book bookToReturn = new Book();
+        JSONParser parser = new JSONParser();
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        String url = "https://openlibrary.org/api/books?bibkeys={isbn}&format=json&jscmd=data";
+        //make a request
+        ResponseEntity<String> response = this.restTemplate
+            .exchange(url, HttpMethod.GET, null, String.class, isbn);
+        //check status ok
+        if (response.getStatusCode() == HttpStatus.OK && !response.getBody().equals("{}")) {
+            // parse response to Json
+            JSONObject json = (JSONObject) parser.parse(response.getBody());
+
+            //read json into tree model
+            JsonNode rootNode = objectMapper.readTree(String.valueOf(json));
+
+            //set properties of book
+            //set genre
+            bookToReturn.setGenre(rootNode.get("ISBN:0385472579").get("subjects").get(0)
```

Comment:
1. Why is this ISBN hardcoded?
2. Instead of writing `rootNode.get("ISBN:0385472579").get("whatever")...` each time you need to access a property, you could just make `JsonNode rootNode = objectMapper.readTree(String.valueOf(json)).get("ISBN:0385472579");`